modules/account-lib/src/coin/sol/instructionParamsFactory.ts

```diff
@@ -155,3 +158,35 @@ function parseStakingActivateInstructions(instructions: TransactionInstruction[]
 
   return instructionData;
 }
+
+/**
+ * Parses Solana instructions to create deactivate tx instructions params
+ * Only supports StakingDeactivate and Memo Solana instructions
+ *
+ * @param {TransactionInstruction[]} instructions - an array of supported Solana instructions
+ * @returns {InstructionParams[]} An array containing instruction params for staking deactivate tx
+ */
+function parseStakingDeactivateInstructions(instructions: TransactionInstruction[]): Array<StakingDeactivate | Memo> {
+  const instructionData: Array<StakingDeactivate | Memo> = [];
+  const deactivateInstruction = StakeInstruction.decodeDeactivate(
+    instructions[stakingDeactivateInstructionsIndexes.Deactivate],
+  );
+  const stakingDeactivate: StakingDeactivate = {
+    type: InstructionBuilderTypes.StakingDeactivate,
+    params: {
+      fromAddress: deactivateInstruction.authorizedPubkey.toString(),
+      stakingAddress: deactivateInstruction.stakePubkey.toString(),
+    },
+  };
+  instructionData.push(stakingDeactivate);
+  const instructionsLength = Object.keys(stakingDeactivateInstructionsIndexes).length;
+  const memoIndex = stakingDeactivateInstructionsIndexes.Memo;
+  if (instructions.length === instructionsLength && instructions[memoIndex]) {
+    const memo: Memo = {
+      type: InstructionBuilderTypes.Memo,
+      params: { memo: instructions[stakingDeactivateInstructionsIndexes.Memo].data.toString() },
+    };
+    instructionData.push(memo);
```

Comment:
Maybe it could be done a common function for parseStakingActivateInstructions and parseStakingDeactivateInstructions to return memo instruction

Reply:
done